src/lib-imap-client/imapc-connection.c

```diff
@@ -382,6 +382,10 @@ static void imapc_connection_set_state(struct imapc_connection *conn,
 
 		conn->selecting_box = NULL;
 		conn->selected_box = NULL;
+
+		if (conn->ips != NULL)
```

Comment:
Hrm, looking at the implementation of it, I may be wrong - it might try to memset the memory first.